modules/gapi/src/backends/fluid/gfluidbackend.cpp

```diff
@@ -1207,35 +1209,39 @@ void GFluidBackendImpl::addBackendPasses(ade::ExecutionEngineSetupContext &ectx)
 
         for (const auto& nh : gim.nodes())
         {
-            if (gim.metadata(nh).get<NodeKind>().k == NodeKind::ISLAND)
+            switch (gim.metadata(nh).get<NodeKind>().k)
+            {
+            case NodeKind::ISLAND:
             {
                 const auto isl = gim.metadata(nh).get<FusedIsland>().object;
                 if (isl->backend() == cv::gapi::fluid::backend())
                 {
-                    // add FluidData to all data nodes inside island
+                    // Add FluidData to all data nodes inside island,
+                    // set internal = true if node is not a slot
```

Comment:
"...is not a slot in terms of higher-level GIslandModel", I'd add for a better clarity.

But it is not necessary for Slot object to be island's start/end points only, (see my above comment)

Reply:
Done